### PR TITLE
Fix dependency parsing

### LINKED ISSUE #606: ⚠️ CI failed ⚠️
[Workflow Run URL](https://github.com/coiled/coiled-runtime/actions/runs/3661707414)

### LINKED ISSUE #607: ⚠️ CI failed ⚠️
[Workflow Run URL](https://github.com/coiled/coiled-runtime/actions/runs/3666559327)

### LINKED ISSUE #608: ⚠️ CI failed ⚠️
[Workflow Run URL](https://github.com/coiled/coiled-runtime/actions/runs/3671491143)

### PR DESCRIPTION
**Pull request by hendrikmakait**
* ~Updates to `coiled>=0.2.54` for `package_sync` fixes.~ [EDIT:] This change has been merged in #609 for visibility.
* Fixes logic that broke after internals of `packaging` changed by using `conda` wherever possible to parse `conda` match specifications.

* Closes #608
* Closes #607 
* Closes #606

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,14 +20,20 @@ def get_requirements():
 
     requirements = {}
     for req in meta["requirements"]["run"]:
-        requirement = Requirement(req)
+        try:
+            requirement = Requirement(req)
+        except InvalidRequirement as e:
```
Comment (hendrikmakait) [inline setup.py:25]: We can't really use `conda` here, so we have to resort to catching the error instead.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ requirements:
   run:
     - python >=3.8,<3.11
     - pip
-    - coiled >=0.2.41
+    - coiled >=0.2.54
```
[the remark this answers is not in the document]
Comment (ncclementi) [reply recipe/meta.yaml:25]: Unrelated, but I wonder if we should then have another patch release to fix this issue.

[the remark this answers is not in the document]
Comment (hendrikmakait) [reply recipe/meta.yaml:25]: #609

**Comment by hendrikmakait** [comment]
CI is green. If we want to have better visibility for the `coiled` update, review and merge #609 first.

**Comment by crusaderky** [comment]
Thank you

**Pull request merged**